src/app/[locale]/(default)/events/[id]/page.tsx

```diff
@@ -0,0 +1,97 @@
+import { getTranslations, setRequestLocale } from 'next-intl/server';
+
+import { Avatar, AvatarImage } from '@/components/ui/Avatar';
+import { Badge } from '@/components/ui/Badge';
+import { Separator } from '@/components/ui/Separator';
+// TODO: Must be replaced with actual events
+import { events } from '@/mock-data/events';
+import { CalendarIcon, MapPinIcon } from 'lucide-react';
+import { notFound } from 'next/navigation';
+
+export async function generateMetadata({
+  params,
+}: {
+  params: Promise<{ locale: string }>;
+}) {
+  const { locale } = await params;
+  const t = await getTranslations({ locale, namespace: 'layout' });
+
+  return {
+    title: t('events'),
+  };
+}
+
+function datesOnSameDay(date1: Date, date2: Date) {
+  return (
+    date1.getUTCFullYear() === date2.getUTCFullYear() &&
+    date1.getUTCMonth() === date2.getUTCMonth() &&
+    date1.getUTCDay() === date2.getUTCDay()
+  );
+}
+
+export default async function EventDetailsPage({
+  params,
+}: {
+  params: Promise<{ locale: string; id: string }>;
+}) {
+  const { locale, id } = await params;
+  setRequestLocale(locale);
+  const t = await getTranslations('ui');
+  const event = events.find((event) => event.id.toString() === id);
+
+  if (!event) return notFound();
+
+  const startDate = new Date(event.startTime);
+  const endDate = new Date(event.endTime);
+  const timeOptions = {
+    hour: '2-digit',
+    minute: '2-digit',
+    hour12: false,
+  } as const;
+
+  const formattedStartTime = startDate.toLocaleTimeString(locale, timeOptions);
```

Comment:
I think using the useFormatter from next-intl with format in range is generally much better.